pymedphys/_dicom/delivery/core.py

```diff
@@ -134,42 +119,68 @@ def _from_dicom_beam(cls, beam, meterset):
         if meterset is None:
             raise ValueError("Meterset should not ever be None")
 
-        leaf_boundaries = beam.BeamLimitingDeviceSequence[-1].LeafPositionBoundaries
+        beam_limiting_device_sequence = beam.BeamLimitingDeviceSequence
+
+        rt_beam_limiting_device_types = {
+            item.RTBeamLimitingDeviceType for item in beam_limiting_device_sequence
+        }
+
+        if rt_beam_limiting_device_types != set(["MLCX", "ASYMY"]):
+            raise ValueError(
+                "Currently only DICOM files that contain "
+                "exactly MLCX and ASYMY are supported. "
+                f"{rt_beam_limiting_device_types} is not supported."
+            )
```

Comment:
So fundamentally the whole `Delivery` pathway assumes only one set of jaws. Honestly I think that means I should move the `Delivery` API out of stable and into `experimental` and flag it to be superseded by something more representative of what is needed.